pkg/machine/keys.go

```diff
@@ -55,17 +55,16 @@ func generatekeys(writeLocation string) error {
 }
 
 // generatekeys creates an ed25519 set of keys
-func generatekeysPrefix(dir string, file string, passThru bool, prefix ...string) error {
+func generatekeysPrefix(location string, passThru bool, prefix ...string) error {
 	args := append([]string{}, prefix[1:]...)
 	args = append(args, sshCommand...)
-	args = append(args, file)
+	args = append(args, location)
```

Comment:
Unfortunately, this should fail in most cases since location is a windows path being passed into a linux command as an argument. This approach could technically work if it was converted to the 9p mount locations on linux, although it's probably easier to go a different route, where the wsl command is changed to an absolute path that we manually lookup ourselves.

Reply:
Just to clarify what I mean is that dir will still work if before creating the command we do exec.LookPath, then convert the result to an absolute path, then pass that value as the command name.

Reply:
You are right. It expects the mount location on linux. 

I agree with your suggestion to use the absolute path and keep dir as it is. See the update PR.